battery/energy_storage_device.py

```diff
@@ -142,6 +142,9 @@ def evolve_one_time_step_constant_current(self, time_step, current):
             err = str(err)
             err = err.split('] ')[-1]
             print("Error when executing dualfoil: ", err)
+        except RuntimeError as err:
+            print(err)
+            raise
```

Comment:
This is not how it is done usually. In your main function, you have a ```try ... catch ...```. When you throw an exception, you catch it in ```catch``` and there you can, for example, print the error message. This way you don't have to copy ```print`` everywhere.